js/wp-seo-shortcode-plugin.js

```diff
@@ -0,0 +1,281 @@
+};
+
+/**
+ * The callback used to replace the shortcodes.
+ *
+ * @param {string} data
+ * @returns {string}
+ */
+YoastShortcodePlugin.prototype.replaceShortcodes = function( data ) {
+	var parsedShortcodes = yoastShortcodePlugin.parsedShortcodes;
+
+	if ( typeof data === 'string' && parsedShortcodes.length > 0 ) {
+		for ( var i in parsedShortcodes ) {
+			data = data.replace( parsedShortcodes[ i ].shortcode, parsedShortcodes[ i ].output );
+		}
+	}
+	return data;
+};
+
+/* DATA SOURCING */
+
+/**
+ * Get data from inputfields and store them in an analyzerData object. This object will be used to fill
+ * the analyzer and the snippetpreview
+ */
+YoastShortcodePlugin.prototype.loadShortcodes = function() {
+	var title = document.getElementById( 'title' ).value;
+	var SEOTitle = document.getElementById( 'yoast_wpseo_title' ).value;
+	var content = this.getContentTinyMCE();
+	var meta = document.getElementById( 'yoast_wpseo_metadesc' ).value;
+
+	this.checkShortcodes( title );
+	this.checkShortcodes( SEOTitle );
+	this.checkShortcodes( content );
+	this.checkShortcodes( meta );
+
+	this.parseShortcodes( this.declareReady );
+};
+
+/**
+ * Bind elements to be able to reload the dataset if shortcodes get added.
+ */
+YoastShortcodePlugin.prototype.bindElementEvents = function() {
+	var elems = [ 'title', 'snippet_title', 'snippet_meta', 'content' ];
+
+	for ( var i = 0; i < elems.length; i++ ) {
+		document.getElementById( elems[ i ] ).addEventListener( 'keydown', yoastShortcodePlugin.inputEventCallback );
+		document.getElementById( elems[ i ] ).addEventListener( 'input', yoastShortcodePlugin.inputEventCallback );
+	}
+};
+
+/**
+ * Callback function for events on input elements. Beware! This function is always called within the scope of the element!
+ *
+ * @param {object} ev The event object
+ */
+YoastShortcodePlugin.prototype.inputEventCallback = function( ev ) {
+	var text;
+
+	if ( ev.currentTarget.id === 'content' ) {
+		text = yoastShortcodePlugin.getContentTinyMCE();
+	}
+	else if ( ev.currentTarget.tagName === 'SPAN' ) {
+		text = ev.currentTarget.textContent;
+	}
+	else {
+		text = ev.currentTarget.value;
+	}
+
+	yoastShortcodePlugin.checkShortcodes( text );
+	yoastShortcodePlugin.parseShortcodes( yoastShortcodePlugin.declareReloaded );
+};
+
+/**
+ * gets content from the content field, if tinyMCE is initialized, use the getContent function to get the data from tinyMCE
+ * @returns {String}
+ */
+YoastShortcodePlugin.prototype.getContentTinyMCE = function() {
+	var val = document.getElementById( 'content' ).value;
+	if ( tinyMCE.editors.length !== 0 ) {
+		val = tinyMCE.get( 'content' ).getContent();
+	}
+
+	return val;
+};
+
+/* SHORTCODE PARSING */
+
+/**
+ * Add a shortcode to the unparsed shortcodes unless was already parsed.
+ *
+ * @param {string} shortcode
+ */
+YoastShortcodePlugin.prototype.checkShortcode = function( shortcode ) {
```

Comment:
Because you are using state here it is harder to follow how this actually works, the function names don't really help in this regard. I would prefer something more direct. Something like this would be much more clear:
```js
var unparsedShortcodes = this.getUnparsedShortcodes( this.getShortcodes() );
if ( 0 < unparsedShortcodes.length ) {
    this.parseShortcodes( unparsedShortcodes );
}
```

That would be placed in the event handler or a new function.